### PR TITLE
Small corrections to configure.

### PR DESCRIPTION
**Pull request by vkni**
I ran `shellcheck` on `configure` script, which gave several warnings. Some of them may be false positive, but it looks like it found several obsolete variables and may be even error.

I checked resulting `configure` script on OSX and Linux (gcc) (`./configure; make world; make clean; make world.opt`).

### CODE REVIEW

File: configure [hunk 1]
```diff
@@ -33,7 +33,6 @@ cpp='cpp'
 asoption=''
 asppoption=''
 cclibs=''
-curseslibs=''
```
Comment (vkni) [inline configure:36]: Line was edited in 2000-10-27.

File: configure [hunk 2]
```diff
@@ -419,10 +418,6 @@ common_cflags=""
 common_cppflags=""
 internal_cflags=""
 internal_cppflags=""
-ocamlc_cflags=""
```
Comment (vkni) [inline configure]: This block is recent - 2017-03-16.

File: configure [hunk 3]
```diff
@@ -543,7 +538,6 @@ case "$cc,$target" in
     exe=".exe"
     ostype="Cygwin";;
   *,*-*-mingw*)
-    dllccompopt="-DCAML_DLL"
```
Comment (vkni) [inline configure:546]: Edited in 2013-06-30, I could not find CAML_DLL anywhere else with git grep.

File: configure [hunk 4]
```diff
@@ -1779,7 +1773,6 @@ if test "$x11_include" = "not found"; then
       if test $dir = /usr/lib; then
         x11_link="-lX11"
       else
-        x11_libs="-L$dir"
```
Comment (vkni) [inline configure:1782]: Edited in 2012-02-01. Variable `x11_libs` can not be found with git grep. Probably here should be `x11_link`, which is defined 4 lines below.

File: configure [hunk 5]
```diff
@@ -1896,12 +1889,9 @@ fi
 
 # Check for mmap support for huge pages and contiguous heap
 if sh ./runtest mmap-huge.c; then
-    has_huge_pages=true
```
Comment (vkni) [inline configure]: This variable is unused now. May be it worth adding it the final report table instead of removing?

File: configure [hunk 6]
```diff
@@ -2187,7 +2177,7 @@ else
   fi
   if $with_spacetime; then
     inf "        spacetime profiling....... yes"
-    if test "$with_spacetime_call_counts" = "true"; then
+    if test "$enable_call_counts" = "true"; then
```
Comment (vkni) [inline configure:2187]: `with_spacetime_call_counts` is not defined anywhere. It seems to me that here should be `enable_call_counts` instead.

File: configure (hunk 1 above)
Comment (dra27) [reply configure:36]: This should have been removed in #1431.

File: configure [hunk 7]
```diff
@@ -419,10 +418,6 @@ common_cflags=""
 common_cppflags=""
 internal_cflags=""
 internal_cppflags=""
-ocamlc_cflags=""
-ocamlc_cppflags=""
-ocamlopt_cflags=""
-ocamlopt_cppflags=""
```
Comment (dra27) [inline configure]: These were introduced in #1114. I don't think removing them is the correct approach, I think they should be being included in the definitions of `OCAMLC_CFLAGS`, `OCAMLC_CPP_FLAGS`, `OCAMLOPT_CFLAGS` and `OCAMLOPT_CPPFLAGS`.

cc @shindere

File: configure (hunk 3 above)
Comment (dra27) [reply configure:546]: This should have been removed in #907.

File: configure (hunk 4 above)
Comment (dra27) [reply configure:1782]: This variable was introduced in 296fc0547052da47bf0e983cab1ce173fa1e8882 when olabl was merged and was necessary for detection of Tcl/Tk - it should have been removed in e82104a755463d481667650ba4f00de535048f39.

File: configure (hunk 5 above)
Comment (dra27) [reply configure]: This was introduced in #297 and has never been used, but I don't think that's a case for its removal.

File: configure (hunk 6 above)
Comment (dra27) [reply configure:2187]: This should have been changed in #1421.

**Review by dra27 — CHANGES_REQUESTED** [review]
Thank you for these - I've annotated them with pull requests/commits where relevant changes were made.

File: configure (hunk 5 above)
Comment (vkni) [reply configure]: Probably @damiendoligez can decide if this variable should be removed or reported at the end of configure run.

**Comment by vkni** [comment]
I removed commits with `ocaml*_*flags` variables and `has_huge_pages`. Now PR contains only lines which had to be removed/changed.

**Review by dra27 — APPROVED** [review]

**Comment by dra27** [comment]
That's great, thank you!

**Pull request merged**